pkg/internal/common/mockoidc/provider.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mockoidc
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// StartProvider a new http endpoint serving a mock OpenID Connect configuration file and returns its base URL.
+// Useful for tests that include code paths that resolve the OIDC configuration file.
+func StartProvider(t *testing.T) string {
+	h := &testOIDCProvider{}
+	srv := httptest.NewServer(h)
+	h.baseURL = srv.URL
+
+	t.Cleanup(srv.Close)
+
+	return srv.URL
+}
+
+type oidcConfigJSON struct {
+	Issuer      string   `json:"issuer"`
+	AuthURL     string   `json:"authorization_endpoint"`
+	TokenURL    string   `json:"token_endpoint"`
+	JWKSURL     string   `json:"jwks_uri"`
+	UserInfoURL string   `json:"userinfo_endpoint"`
+	Algorithms  []string `json:"id_token_signing_alg_values_supported"`
+}
+
+type testOIDCProvider struct {
+	baseURL string
+}
+
+func (t *testOIDCProvider) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	response, err := json.Marshal(&oidcConfigJSON{
+		Issuer:      t.baseURL,
+		AuthURL:     fmt.Sprintf("%s/oauth2/auth", t.baseURL),
+		TokenURL:    fmt.Sprintf("%s/oauth2/token", t.baseURL),
+		JWKSURL:     fmt.Sprintf("%s/oauth2/certs", t.baseURL),
+		UserInfoURL: fmt.Sprintf("%s/oauth2/userinfo", t.baseURL),
+		Algorithms:  []string{"RS256"},
```

Comment:
not sure why this is hard coded? are we only supporting RS256?

Reply:
@Baha-sk I didn't add this... I was just shuffling lines around

Reply:
@Baha-sk field defined here: https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

it's the `id_token_signing_alg_values_supported` claim:

> REQUIRED. JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP for the ID Token to encode the Claims in a JWT [JWT]. The algorithm RS256 MUST be included. The value none MAY be supported, but MUST NOT be used unless the Response Type used returns no ID Token from the Authorization Endpoint (such as when using the Authorization Code Flow). 

When I originally added this I just put in whatever algorithms supported by go-oidc: https://github.com/coreos/go-oidc/blob/55058f587227e7aec3703e74dcbdabb75131481f/oidc.go#L94-L107.

Reply:
Please keep in mind this object is just for unit tests... nothing more.

Reply:
I didn't notice the `mock` word in patch... it's ok then.